### ISSUE TITLE
Build task surrounds args with asterisks in them in single quotes. 

### ISSUE THREAD

**Issue by datanasio** [microsoft/vscode-cpptools#12023]
### Environment

- OS and Version: Ubuntu Debian and Mac OS
- VS Code Version: 1.86.2
- C/C++ Extension Version: 1.19.4
- If using SSH remote, specify OS of remote machine: 

### Bug Summary and Steps to Reproduce

Bug Summary:
If an asterisk if used in the tasks.json file for compiler args then the line is enclosed in single quotes.
The compiler reports that the file doesn't exist because the wildcard is not expanded.

Steps to reproduce:
1. On a linux or mac computer, change the args line in tasks.json to read, "${fileDirname}/*.cpp"
2. Use F5 or the run/degug button to build and execute the code.
4. The compiler reports that the file cannot be found.
    This is the code that's produced to build the executable:
     /usr/bin/g++ "/home/dom/test/*.cpp" -o /home/dom/test/test.out
    Error: cc1plus: fatal error: /home/dom/test/*.cpp: No such file or directory

### Configuration and Logs

```shell
{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: g++ build active file",
            "command": "/usr/bin/g++",
            "args": [
                "-fdiagnostics-color=always",
                "-g",
                "${fileDirname}/*.cpp",
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}.out"
            ],
            "options": {
                "cwd": "${fileDirname}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "detail": "Task generated by Debugger."
        }
    ],
    "version": "2.0.0"
}
```

### Other Extensions

none

**Comment by sean-mcmanus** [comment]
We're already working on a fix for this with https://github.com/microsoft/vscode-cpptools/issues/12001 .

Downgrading to 1.18.5 is the recommended workaround until we can release an update.

**Comment by datanasio** [comment]
Thanks, I posted that fix on the VSCode site. Thank you for letting me know though. I didn't see the other report.


[Mt. San Antonio College (Mt. SAC) Logo]
Dominick Atanasio
Professor of Computer Science
Math and Computer Science Dept.
Natural Sciences Div.
[Office]        Bldg. 61 Room 1656
[Email] ***@***.******@***.***>

Mt. San Antonio College
1100 N. Grand Ave.,
Walnut CA 91789
www.mtsac.edu</>
Fingerprint=    1DC3 0A5F 7453 0048 D662 AC4E D432 2679 D4A5 B02B
On 2/26/24 1:34 PM, Sean McManus wrote:

You don't often get email from ***@***.******@***.***>. Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

  EXTERNAL SENDER - Exercise caution with requests, links, and attachments.

We're already working on a fix for this with #12001<https://github.com/microsoft/vscode-cpptools/issues/12001> .

Downgrading to 1.18.5 is the recommended workaround until we can release an update.
